### ISSUE TITLE
Calculate NLL

### ISSUE THREAD

**Issue by XJWang0** [airalcorn2/deformer#2]
I'm sorry to bother you again. I want to know how NLL in this article is calculated. Is it the same as loss in the code, or is it obtained by loss through some kind of operation? And “taken over 10 orders ” means？

**Comment by airalcorn2** [comment]
>Is it the same as loss in the code

Yes, the NLL is the same as the loss. You can see how I evaluated the model on the MINST test set [here](https://github.com/airalcorn2/deformer/blob/master/paper_functions.py#L16).

>And “taken over 10 orders ” means

Because the DEformer is only _approximately_ permutation invariant, the NLL for a given sample will vary depending on the order of the features. So the values reported in the paper are the average NLLs over 10 random orders for each sample.

**Comment by XJWang0** [comment]
So I just run train_ deformer.py,  the _test_ loss_ best_ valid_ is the NLL? Or need to run _get_multi_order_nlls_ in paper_functions.py?

**Comment by XJWang0** [comment]
My email address is Wxj_1010@outlook.com, please contact me if you see it. Thanks !

**Comment by airalcorn2** [comment]
You need to run the `get_multi_order_nlls` function I linked to in `paper_functions.py`. `test_loss_best_valid` is the average NLL when only using the raster scan order for each test sample, and it's only there for monitoring purposes.

**Comment by XJWang0** [comment]
What was your original `test_loss_best_valid`? The result I got was about 0.14, and the corresponding NLL was about 83. Is this related to the batch size?

**Comment by airalcorn2** [comment]
'test_loss_best_valid' is the average NLL per pixel, whIle 'get_multi_order_nlls' calculates the average NLL per image, i.e., it's summing the NLL over the pixels for each image.

**Comment by airalcorn2** [comment]
Unfortunately, I don't remember my original `test_loss_best_valid`. [The batch size is indeed often an important hyperparameter when training models.](https://arxiv.org/abs/1803.09820)
